substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ResourceConfigurationParser.java

```diff
@@ -48,21 +48,40 @@ public void parseAndRegister(Reader reader) throws IOException {
     }
 
     private void parseTopLevelObject(Map<String, Object> obj) {
-        Object resourcesObject = null;
+        Map<String, Object> resourcesObject = null;
         Object bundlesObject = null;
         for (Map.Entry<String, Object> pair : obj.entrySet()) {
             if ("resources".equals(pair.getKey())) {
-                resourcesObject = pair.getValue();
+                resourcesObject = asMap(pair.getValue(), "Attribute 'resources' must be an object");
             } else if ("bundles".equals(pair.getKey())) {
                 bundlesObject = pair.getValue();
             } else {
                 throw new JSONParserException("Unknown attribute '" + pair.getKey() + "' (supported attributes: name) in resource definition");
             }
         }
         if (resourcesObject != null) {
-            List<Object> resources = asList(resourcesObject, "Attribute 'resources' must be a list of resources");
-            for (Object object : resources) {
-                parseEntry(object, "pattern", registry::addResources, "resource descriptor object", "'resources' list");
+            Object includesObject = null;
+            Object excludesObject = null;
+            for (Map.Entry<String, Object> pair : resourcesObject.entrySet()) {
```

Comment:
That's a breaking change. We can't do that. Please also support old-style `"includes"`.

Reply:
Yes, you are right. I changed the code to ensure backward compatibility.